src/Handler/StreamHandler.php

```diff
@@ -277,6 +278,7 @@ private function createStream(RequestInterface $request, array $options)
 
         $params = [];
         $context = $this->getDefaultContext($request, $options);
+        $uri = $request->getUri()->withFragment('');
```

Comment:
See my comment below on why I created a separate `$uri` variable outside the callback.